src/main/java/essencepouchtracking/EssencePouchTrackingPlugin.java

```diff
@@ -531,9 +555,44 @@ public void onItemContainerChanged(ItemContainerChanged itemContainerChanged)
 						this.handPouchActionsPostDegrade(currentPouch);
 					}
 				}
+
+				// Check to see if the player has an Abyssal Lantern (Redwood) in their inventory
+				if (itemId == ItemID.ABYSSAL_LANTERN_REDWOOD_LOGS)
+				{
+					this.hasRedwoodAbyssalLantern = true;
+					log.debug("Player added a Redwood Lantern in their inventory.");
+				}
+			}
+
+			if (removedItems.contains(ItemID.ABYSSAL_LANTERN_REDWOOD_LOGS))
+			{
+				this.hasRedwoodAbyssalLantern = false;
+				log.debug("Player removed the Redwood Lantern from their inventory.");
 			}
 			this.previousInventory = currentInventory;
 		}
+		else if (itemContainerChanged.getContainerId() == InventoryID.EQUIPMENT.getId())
+		{
+			Item capeItem = this.getEquipmentContainer().getItem(EquipmentInventorySlot.CAPE.getSlotIdx());
+			boolean hasEquippedSpecialCape = capeItem != null && this.RC_CAPES_SET.contains(capeItem.getId());
```

Comment:
and then instead of seeing if the item is found inside the cape set, we could simply just see if the cape's name `starts with runecrafting cape || ends with max cape`.